app/api/api/serializers.py

```
@@ -168,18 +245,54 @@ def validate_data_dictionary_file(self, value):
                         )
                     )
 
+            # Collect csv_file_name if line exists
+            if line and len(line) > 0:
+                csv_file_names.add(line[0].strip())
+
+        # Log the collected csv_file_names and sr_table_names
+        logger.info(f"CSV file names: {csv_file_names}")
+        logger.info(f"Scan Report table names: {self.sr_table_names}")
+
+        # Checking for mismatches between csv_file_name entries and expected table names:
+        logger.info("Checking for matching table names of DD and SR.")
+
+        if not hasattr(self, "sr_table_names") or not csv_file_names:
+            missing = (
+                "Scan Report table names"
+                if not hasattr(self, "sr_table_names")
+                else "CSV file names"
+            )
+            logger.error(f"{missing} are missing for validation.")
+            raise ValueError(f"{missing} are required for validation.")
+
+        # Validate each csv_name against sr_table_names
+        for csv_name in csv_file_names:
+            if csv_name not in self.sr_table_names:
+                errors.append(
+                    ParseError(
+                        f"CSV name in Data Dictionary '{csv_name}' is not matching any table in Scan Report. "
+                        f"Available tables: {', '.join(sorted(self.sr_table_names))}"
+                    )
+                )
+
         if errors:
             raise ParseError(errors)
 
+        logger.info("Data dictionary file is valid and ready for upload.")
+        data_dictionary.seek(0)
```

Do we need this?

---

Yes. It is helpful in silent failures and resets the file pointer to the start before upload.